ipn/ipnlocal/local.go

```diff
@@ -2358,7 +2360,16 @@ func (b *LocalBackend) enterState(newState ipn.State) {
 		if authURL == "" {
 			systemd.Status("Stopped; run 'tailscale up' to log in")
 		}
-	case ipn.Starting, ipn.NeedsMachineAuth:
+	case ipn.Starting:
+		if oldState == ipn.NeedsLogin && !keyExpired && blocked {
+			// This means that key was extended **after** expiry,
+			// unblock engine updates.
+			b.blockEngineUpdates(false)
+		}
+		b.authReconfig()
+		// Needed so that UpdateEndpoints can run
```

Comment:
it's not obvious what line (prior or following) this comment is referring to, or why they make UpdateEndpoints run?

same on line 2374

Reply:
I honestly have no idea what it does/means either, but I'm not changing that code now so leaving it as is.